File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/chooser/ChooserSubcomponent.java
package com.tevinjeffrey.rutgersct.ui.chooser;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface ChooserSubcomponent extends AndroidInjector<ChooserFragment> {

  void inject(ChooserPresenterImpl chooserPresenter);

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<ChooserFragment> {
    @Override public abstract ChooserSubcomponent build();
  }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/chooser/ChooserSubcomponent.kt
package com.tevinjeffrey.rutgersct.ui.chooser

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface ChooserSubcomponent : AndroidInjector<ChooserFragment> {

  fun inject(viewModel: ChooserViewModel)

  @Subcomponent.Builder
  abstract class Builder : AndroidInjector.Builder<ChooserFragment>() {
    abstract override fun build(): ChooserSubcomponent
  }
}
